paws.common/src/encoding.cpp

```diff
@@ -0,0 +1,111 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// This encoder has been modified from the excellent urltools R package
+// https://github.com/Ironholds/urltools/blob/master/src/encoding.cpp
+
+std::string to_hex(char x){
+
+  //Holding objects and output
```

Comment:
Nit: I feel like there should be a shorter, easier to read way to convert using `sprintf` or something but I'm not able to get it easily so not really important.